firefox_puppeteer/base.py

```diff
@@ -22,3 +23,21 @@ def marionette(self):
 
     def get_marionette(self):
         return self.marionette
+
+
+class UIBaseLib(BaseLib):
+    """A base class for all UI element wrapper classes inside a chrome window."""
+
+    def __init__(self, marionette_getter, window):
+        BaseLib.__init__(self, marionette_getter)
+
+        # importing globally doesn't work
+        from .ui.windows import BaseWindow
```

Comment:
Using `type()` here will not work given that this will not cover all the window sub-classes. I would leave it for now and file a follow-up bug to sort this out.